sql/resource_sql_alerts.go

```diff
@@ -0,0 +1,143 @@
+package sql
+
+import (
+	"context"
+
+	"github.com/databricks/databricks-sdk-go/service/sql"
+	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+type AlertOptions struct {
+	Column        string `json:"column"`
+	Op            string `json:"op"`
+	Value         string `json:"value"`
+	Muted         bool   `json:"muted,omitempty" tf:"suppress_diff"`
+	CustomBody    string `json:"custom_body,omitempty"`
+	CustomSubject string `json:"custom_subject,omitempty"`
+}
+
+type AlertEntity struct {
+	Name    string        `json:"name"`
+	QueryId string        `json:"query_id"`
+	Rearm   int           `json:"rearm,omitempty"`
+	Options *AlertOptions `json:"options"`
+	Parent  string        `json:"parent,omitempty" tf:"suppress_diff,force_new"`
+}
+
+func (a *AlertEntity) toCreateAlertApiObject(s map[string]*schema.Schema, data *schema.ResourceData) (sql.CreateAlert, error) {
+	common.DataToStructPointer(data, s, a)
+
+	var ca sql.CreateAlert
+	ca.Name = a.Name
+	ca.Parent = a.Parent
+	ca.QueryId = a.QueryId
+	ca.Rearm = a.Rearm
+	ca.Options = sql.AlertOptions{
+		Column:        a.Options.Column,
+		CustomBody:    a.Options.CustomBody,
+		CustomSubject: a.Options.CustomSubject,
+		Muted:         a.Options.Muted,
+		Op:            a.Options.Op,
+		Value:         a.Options.Value,
+	}
+
+	return ca, nil
+}
+
+func (a *AlertEntity) toEditAlertApiObject(s map[string]*schema.Schema, data *schema.ResourceData) (sql.EditAlert, error) {
+	common.DataToStructPointer(data, s, a)
+
+	return sql.EditAlert{
+		AlertId: data.Id(),
+		Name:    a.Name,
+		Options: sql.AlertOptions{
+			Column:        a.Options.Column,
+			CustomBody:    a.Options.CustomBody,
+			CustomSubject: a.Options.CustomSubject,
+			Muted:         a.Options.Muted,
+			Op:            a.Options.Op,
+			Value:         a.Options.Value,
+		},
+		QueryId: a.QueryId,
+		Rearm:   a.Rearm,
+	}, nil
+}
+
+func (a *AlertEntity) fromAPIObject(apiAlert *sql.Alert, s map[string]*schema.Schema, data *schema.ResourceData) error {
+	a.Name = apiAlert.Name
+	a.Parent = apiAlert.Parent
+	a.QueryId = apiAlert.Query.Id
+	a.Rearm = apiAlert.Rearm
+	a.Options = &AlertOptions{
+		Column:        apiAlert.Options.Column,
+		Op:            apiAlert.Options.Op,
+		Value:         apiAlert.Options.Value,
+		Muted:         apiAlert.Options.Muted,
+		CustomBody:    apiAlert.Options.CustomBody,
+		CustomSubject: apiAlert.Options.CustomSubject,
+	}
+
+	return common.StructToData(a, s, data)
+}
+
+func ResourceSqlAlert() *schema.Resource {
+	s := common.StructToSchema(AlertEntity{}, func(m map[string]*schema.Schema) map[string]*schema.Schema {
```

Comment:
can you check if we can do it without `to/fromAPIObject`? `ForceNew` you can add via this schema callback for sure. suppress diff - not quite sure if we need it.